From: Tim Bird <tim.bird@sony.com>
Subject: mm: Add SPDX id lines to some mm source files
Date: Wed, 4 Feb 2026 14:31:01 -0700

Some of the memory management source files are missing
SPDX-License-Identifier lines.  Add appropriate IDs
to these files (mostly GPL-2.0, but one LGPL-2.1).

Link: https://lkml.kernel.org/r/20260204213101.1754183-1-tim.bird@sony.com
Signed-off-by: Tim Bird <tim.bird@sony.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/dmapool_test.c   |    1 +
 mm/folio-compat.c   |    1 +
 mm/gup_test.c       |    1 +
 mm/hugetlb_cgroup.c |    9 +--------
 mm/memfd.c          |    3 +--
 mm/mmu_gather.c     |    1 +
 mm/rmap.c           |    2 +-
 mm/shmem.c          |    3 +--
 8 files changed, 8 insertions(+), 13 deletions(-)

--- a/mm/dmapool_test.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/dmapool_test.c
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
 #include <linux/device.h>
 #include <linux/dma-map-ops.h>
 #include <linux/dma-mapping.h>
--- a/mm/folio-compat.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/folio-compat.c
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
 /*
  * Compatibility functions which bloat the callers too much to make inline.
  * All of the callers of these functions should be converted to use folios
--- a/mm/gup_test.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/gup_test.c
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
 #include <linux/kernel.h>
 #include <linux/mm.h>
 #include <linux/slab.h>
--- a/mm/hugetlb_cgroup.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/hugetlb_cgroup.c
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: LGPL-2.1
 /*
  *
  * Copyright IBM Corporation, 2012
@@ -7,14 +8,6 @@
  * Copyright (C) 2019 Red Hat, Inc.
  * Author: Giuseppe Scrivano <gscrivan@redhat.com>
  *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms of version 2.1 of the GNU Lesser General Public License
- * as published by the Free Software Foundation.
- *
- * This program is distributed in the hope that it would be useful, but
- * WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- *
  */
 
 #include <linux/cgroup.h>
--- a/mm/memfd.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/memfd.c
@@ -1,10 +1,9 @@
+// SPDX-License-Identifier: GPL-2.0
 /*
  * memfd_create system call and file sealing support
  *
  * Code was originally included in shmem.c, and broken out to facilitate
  * use by hugetlbfs as well as tmpfs.
- *
- * This file is released under the GPL.
  */
 
 #include <linux/fs.h>
--- a/mm/mmu_gather.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/mmu_gather.c
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
 #include <linux/gfp.h>
 #include <linux/highmem.h>
 #include <linux/kernel.h>
--- a/mm/rmap.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/rmap.c
@@ -1,8 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0
 /*
  * mm/rmap.c - physical to virtual reverse mappings
  *
  * Copyright 2001, Rik van Riel <riel@conectiva.com.br>
- * Released under the General Public License (GPL).
  *
  * Simple, low overhead reverse mapping scheme.
  * Please try to keep this thing as modular as possible.
--- a/mm/shmem.c~mm-add-spdx-id-lines-to-some-mm-source-files
+++ a/mm/shmem.c
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0
 /*
  * Resizable virtual memory filesystem for Linux.
  *
@@ -17,8 +18,6 @@
  *
  * tiny-shmem:
  * Copyright (c) 2004, 2008 Matt Mackall <mpm@selenic.com>
- *
- * This file is released under the GPL.
  */
 
 #include <linux/fs.h>
_
